engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1203,17 +1209,33 @@ private AllocateResponse sendContainerAskToRM(List<ContainerRequest> containerRe
       amRmClient.releaseAssignedContainer(containerId);
     }
 
+    return amRmClient.allocate(0);
+  }
+
+  private void processContainerStopRequests()
```

Comment:
Is there a specific need for refactoring this logic out of sendContainerAskToRM into a separate method. The sendContainerAskToRM method already seems to be handling all the different kinds of container requests.